src/include/cell_inline.h

```diff
@@ -986,12 +990,16 @@ __cell_kv_window_cleanup(WT_SESSION_IMPL *session, WT_CELL_UNPACK_KV *unpack_kv)
     if (unpack_kv != NULL) {
         tw = &unpack_kv->tw;
         if (tw->start_txn != WT_TXN_NONE) {
-            tw->start_txn = WT_TXN_NONE;
-            F_SET(unpack_kv, WT_CELL_UNPACK_TIME_WINDOW_CLEARED);
+            if (!F_ISSET(session, WT_SESSION_DEBUG_DO_NOT_CLEAR_TXN_ID)) {
```

Comment:
Personally I am thinking that the code readability can be abit confusing, since it acts as a double negative. @kommiharibabu I was thinking if it is possible to place this if statement and move it to the beginning of the function, and check if it is set, then we return from the function directory. What are your thoughts? Will this incur any side effects?

Reply:
It is a good idea @jiechenbo. It doesn't cause any problems. I prefer to move the check to the caller function `__cell_unpack_window_cleanup` and decide whether to call the cleanup functions or not completely.

Reply:
Yeah, that makes sense, however, I'm not sure how to avoid using a double negative since the default is to clear the txn_id's. I've moved the flag check `__cell_unpack_window_cleanup` which is definitely cleaner.

Reply:
I have looked at the new code, I think you can just return early from the function if the flag is set, and that should work.

Reply:
Good idea, fixed.